src/librbd/api/Mirror.cc

```diff
@@ -399,18 +392,6 @@ int Mirror<I>::image_enable(I *ictx, bool relax_same_pool_parent_check) {
     }
   }
 
-  if (!ictx->test_features(RBD_FEATURE_JOURNALING)) {
```

Comment:
How can we detect we are using (want to use) journaling mode at this stage? I though that when we enable mirroring for an image without journaling feature enabled, that means we want snapshot mirroring.

Otherwise we would need to pass a param to `mirrror enable` request which mode we want. Do you want it this way?

Reply:
Yes, we want to be explicit about the mode -- which would require a new API. To keep this PR simple (it's getting very long), perhaps just add a new "rbd_default_mirror_image_mode" param and we can swing back to extend the API + CLI later?